web/src/components/Upload/util.ts

```diff
@@ -1,43 +1,132 @@
 import { FileWithPath } from 'react-dropzone'
+import moment from 'moment-timezone'
+import { DustrakMeta, ValidateMeta } from 'components/Upload/types'
 
-// export const parseDatesFromCsv = (file: FileWithPath): Array<string> => {}
+export const getGpsStart = (textLines: Array<string>): moment.Moment => {
+  for (const line of textLines) {
+    if (line.startsWith('$GPRMC')) {
+      const lineFields: Array<string> = line.split(',', 10)
+      const encodedTime: string = lineFields[1]
+      const encodedDate: string = lineFields[9]
+      const endTime: moment.Moment = moment.utc(
+        `${encodedDate} ${encodedTime}`,
+        'DDMMYYYY hh mm ss.SS'
+      )
+      return endTime
+    }
+  }
+  return moment.utc('')
+}
 
-const parseDateFromLogFileName = (fileName: string): string => {
-  const fileNameArr = fileName.split('_')
-  const dateString = fileNameArr[1]
-  return `${dateString.slice(0, 4)}/${dateString.slice(
-    4,
-    6
-  )}/${dateString.slice(6)}`
+export const getDustrakStart = (textLines: Array<string>): moment.Moment => {
+  try {
+    const startTime: string = textLines[6].split(',')[1]
+    const startDate: string = textLines[7].split(',')[1]
+    const startDatetime: moment.Moment = moment.tz(
+      `${startDate} ${startTime}`,
+      'MM-DD-YYYY hh:mm:ss a',
+      'America/Los_Angeles'
+    )
+    return startDatetime
+  } catch (e) {
+    if (e instanceof TypeError) {
+      console.warn(e)
+      return moment('')
+    } else {
+      throw e
+    }
+  }
 }
 
-export const validateFiles = (files: Array<FileWithPath>): string => {
-  if (files.length > 0 && files.length < 2) {
-    return 'We need one GPS log file and one DusTrak cvs file. Please add a file to continue.'
+export const getDustrakEnd = (
+  textLines: Array<string>,
+  startDatetime: moment.Moment
+): moment.Moment => {
+  try {
+    const endDatetime: moment.Moment = startDatetime.clone()
+    const testLength: string = textLines[8].split(',')[1]
+    const [days, hours, minutes]: Array<string> = testLength.split(':')
+    endDatetime
+      .add(days, 'days')
+      .add(hours, 'hours')
+      .add(minutes, 'minutes')
+    return endDatetime
+  } catch(e){
+    if (e instanceof TypeError){
+      console.warn(e)
+      return moment('')
+    } else{
+      throw e
+    }
   }
+}
 
-  if (files.length > 2) {
-    return 'We need exactly one GPS log file and one DusTrak cvs file. Please remove additional files.'
+export const validateFiles = async (
+  files: Array<FileWithPath>
+): Promise<ValidateMeta> => {
+  const dustrakData: DustrakMeta = {
+    startDatetime: '',
+    endDatetime: ''
+  }
+  const validateData: ValidateMeta = {
+    message: '',
+    dustrakMeta: dustrakData
   }
 
-  if (files.length === 2) {
-    let logFile: FileWithPath | undefined
-    let csvFile: FileWithPath | undefined
+  if (files.length > 0 && files.length < 2) {
```

Comment:
overall I think we might want to take a more "composable" approach to writing these series of validations and avoid if {} else {} statements if possible. I think to do that, it might make more sense to just early return as soon as a validation fails? 

That way if we ever need to add or remove validations down the road, it would be a simpler process of just deleting one conditional rather than having to restructure conditional logic.

Reply:
**Response**
I'll revert
**Context**
The `if/else`s are coupled with the idea of having an object that tracks the state of validated times- there's a single return that has the final state of the whole object. But, I'm reverting back to only returning a string and might as well revert to a `if condition then return message` paradigm.